site/content/concepts/device-notifications.md

```diff
@@ -17,86 +17,65 @@ Please join our mailing list or consult the IRC channel in case you have any que
-
-In the following it is described how a *Business application* can be notified that a device is ready to receive a command. 
-This is referred to as `device command readiness`.
+This is the purpose of the following *Time until disconnect notification*.
 
-## Device command readiness notification
+## Time until disconnect notification
 
-To notify an application about the command readiness of a device specific properties are set in a downstream message.
+This notification can be triggered in an application by any downstream message as long as the following AMQP 1.0 properties are set:
 
-This can be done
+- the property `creation-time` (automatically set by protocol adapters to the current time when creating the message)
+- an additional application-property `ttd` (the number of seconds the device will stay connected to the protocol adapter)
 
-- implicitly: when sending a downstream message for any reason (e.g. telemetry or event message)
-- explicitly: by sending a specific empty event (e.g. if a device does not have any other downstream message to send)
+A device that wants to only trigger this notification (without sending other downstream data) may send an 
+*empty notification* event.
 
+The application can register a callback for this notification that is invoked if the contained
+timestamp is not expired at the time the message was received. The callback provides an instance of `TimeUntilDisconnectNotification`
+that contains all necessary data to let the application send an upstream message to the device.
 
-The following AMQP 1.0 properties of the message are relevant for the notification:
+See Hono's example module for details where such a notification callback is used.
 
-- the AMQP 1.0 property `creation-time` (which is automatically set by protocol adapters)
-- an additional application-property `ttd` (containing the amount of seconds the device stays connected) in the AMQP 1.0 message
  
 Please refer to the [Telemetry API]({{< relref "api/Telemetry-API.md" >}}) and the [Event API]({{< relref "api/Event-API.md" >}}) for further details.
 
-### Time based validation of notification 
+### Time based validation of a notification 
 
 An application receiving a downstream message containing the notification relevant properties can verify at any point in
-time if a device is ready to receive a command by:
+time if a device is ready to receive an upstream message by:
 
-- adding the values of the `creation-time` and the `ttd` properties to build a *valid until timestamp*
-- comparing the current time with the *valid until timestamp*
+- adding the values of the `creation-time` and the `ttd` properties to build an *expiration timestamp*
+- comparing the current time with the *expiration timestamp*
 
-If the current time is *before* the *valid until timestamp*, the notification shall be regarded as being currently valid.
+If the current time is *before* the *expiration timestamp*, the notification shall be regarded as being expired.
```

Comment:
My understanding is that the notification is expired if the current time is **after** the expiration time.